run-tests.pl

```diff
@@ -645,8 +645,8 @@ sub _push_test
    }
 
    if( exists $params{implementation_specific} ) {
-       if ( $HS_FACTORY->implementation_name() ne $params{implementation_specific} ) {
-           return;
+       unless (grep { $_ eq $HS_FACTORY->implementation_name() } @{$params{implementation_specific}}) {
```

Comment:
also:

```suggestion
       unless (grep { $_ eq $HS_FACTORY->implementation_name() } @{ $params{implementation_specific} }) {
```

because leo.